account_fiscal_product_rule/models/account_fiscal_position.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2022 Akretion France (http://www.akretion.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class AccountFiscalPositionProductRule(models.Model):
+    _name = "account.fiscal.position.product.rule"
+    _description = "Account Fiscal Position Rule in Product"
+
+    name = fields.Char(required=True)
+    fiscal_position_id = fields.Many2one("account.fiscal.position", required=True)
+    product_tmpl_ids = fields.Many2many("product.template")
+    product_category_ids = fields.Many2many("product.category")
+    account_income_id = fields.Many2one("account.account")
+    account_expense_id = fields.Many2one("account.account")
+    seller_tax_ids = fields.Many2many("account.tax", "account_tax_seller")
+    supplier_tax_ids = fields.Many2many(
+        "account.tax",
+        "account_tax_supplier",
+    )
+    company_id = fields.Many2one(related="fiscal_position_id.company_id")
+
+
+class AccountFiscalPosition(models.Model):
+    _inherit = "account.fiscal.position"
+
+    fiscal_position_product_rule_ids = fields.One2many(
+        "account.fiscal.position.product.rule",
+        "fiscal_position_id",
+        string="Product Fiscal Rules",
+    )
+
+    def map_tax(self, taxes, product=None, partner=None):
+        if product or self.env.context.get("product_id", False):
+            if not product:
+                product = self.env["product.product"].browse(
+                    self.env.context.get("product_id", False)
+                )
+            for fp in self:
+                fiscal_product_rules = (
```

Comment:
you can rename rule instead of fiscal_product_rules
```suggestion
                rule = (
```

Note we should have only one rule (see python constraint in product.py)